### PR TITLE
Readme update

### PR DESCRIPTION
**Pull request by dlepow**
## Purpose

* Update to root Readme.md. Check whether install available from Releses.

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ x] Documentation content changes
- [ ] Other... Please describe:

## Does this introduce a breaking change?

- [ ] Yes
- [x ] No

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,24 +1,140 @@
...
+
+- API version sets, APIs, and all child resources, including versions, revisions, and policies 
+
+- Named values 
+
+- Products 
+
+- Subscriptions 
+
+The tool aborts migration if it detects:
```
Comment (mikebudzynski) [inline README.md]: "if it detects the following scenarios unsupported in workspaces preview"

File: README.md [hunk 2]
```diff
@@ -1,24 +1,140 @@
...
+    TBD
+    ``` 
+
+1. Select the API Management instance in which to perform migration.
+
+1. From the list that's displayed, select the APIs or API version sets to migrate. 
+
+1. Select a workspace to migrate the resources to. 
+
+    The tool analyzes dependencies of selected APIs, like named values or subscriptions. See [Supported dependencies](#supported-dependencies) for the list of resources supported in the tool and [Roadmap](#roadmap) for the list of resources that are planned for future support.
```
Comment (mikebudzynski) [inline README.md]: How about:

* Make this part (starting line 61) a new point and talk about the migration plan here
* Change the next point to "Confirm to execute the migration."

File: README.md [hunk 3]
```diff
@@ -1,24 +1,140 @@
...
+1. Open the terminal window and log in with Azure CLI: 
+
+    ```  
+    az login
+    ``` 
+
+1. Run the tool with the following command:
+
+    ```
+    TBD
```
Comment (mikebudzynski) [inline README.md]: I imagine this would also include dotnet commands to install software dependencies and compile it

File: README.md [hunk 4]
```diff
@@ -1,24 +1,140 @@
...
+
+* A backend used in the `set-backend` policy 
+
+* Assignment of an API to a self-hosted gateway 
+
+Other dependent resources aren't currently supported but are planned for future support. See [Roadmap](#roadmap).
+
+## Prerequisites
+
+- [.NET 6.0](https://dotnet.microsoft.com/download/dotnet/6.0)
```
Comment (mikebudzynski) [inline README.md]: @maciejtreder, is this .NET or .NET Core? Is Windows a prerequisite?

File: README.md [hunk 5]
```diff
@@ -1,24 +1,140 @@
...
+
+    The tool analyzes dependencies of selected APIs, like named values or subscriptions. See [Supported dependencies](#supported-dependencies) for the list of resources supported in the tool and [Roadmap](#roadmap) for the list of resources that are planned for future support. 
+    
+    1. Manually check if your API uses any unsupported dependencies before attempting the migration; if it does, your API can't be fully migrated with the tool. 
+
+    1. If the tool detects a dependency of a type that isn't supported in workspaces, it will automatically abort the migration attempt.  
+
+1. The tool analyzes the migration plan. Confirm to execute the migration. 
+
+1. The tool copies the selected APIs, API version sets, and revisions into the selected workspace. It appends a random hash to the API name and API suffix to avoid conflicts.
```
Comment (mikebudzynski) [inline README.md]: The hash is the same for the whole migration, so we can change this text to something like "It computes a random hash and appends it to the API name and API suffix (...)"

File: README.md [hunk 6]
```diff
@@ -1,24 +1,140 @@
...
+
+1. The tool copies the selected APIs, API version sets, and revisions into the selected workspace. It appends a random hash to the API name and API suffix to avoid conflicts. 
+
+    For each detected API dependency: 
+
+    * If the dependency is used *only* by APIs that are being migrated to the workspace, the tool copies it to the workspace. In the future, the original resource on the service level will be deleted. While in preview, the tool keeps all the original resources for backup. 
+
+    * If the dependency is used also by other APIs and it's of a type that allows referencing a service-level resource from a workspace-level resource (that is, product or tag), this dependency along with all of its dependencies (for example, subscription to a product or tag) will remain on the service level and the migrated API will reference it from a workspace. 
+
+    * If the dependency is used also by other APIs and doesn't allow referencing a service-level resource from a workspace-level resource, this dependency will be copied to a workspace with a random hash appended to its name.
```
Comment (mikebudzynski) [inline README.md]: * *the*, not *a*, hash appended to its name
* "The future iterations of the tool will still leave it undeleted."

File: README.md [hunk 7]
```diff
@@ -1,24 +1,140 @@
...
+
+    1. If the tool detects a dependency of a type that isn't supported in workspaces, it will automatically abort the migration attempt.  
+
+1. The tool analyzes the migration plan. Confirm to execute the migration. 
+
+1. The tool copies the selected APIs, API version sets, and revisions into the selected workspace. It appends a random hash to the API name and API suffix to avoid conflicts. 
+
+    For each detected API dependency: 
+
+    * If the dependency is used *only* by APIs that are being migrated to the workspace, the tool copies it to the workspace. In the future, the original resource on the service level will be deleted. While in preview, the tool keeps all the original resources for backup.
```
Comment (mikebudzynski) [inline README.md]: I believe we also append hash to the name in this case, but we will change this behavior in the future.

File: README.md [hunk 8]
```diff
@@ -1,24 +1,140 @@
...
+    
+    1. Manually check if your API uses any unsupported dependencies before attempting the migration; if it does, your API can't be fully migrated with the tool. 
+
+    1. If the tool detects a dependency of a type that isn't supported in workspaces, it will automatically abort the migration attempt.  
+
+1. The tool analyzes the migration plan. Confirm to execute the migration. 
+
+1. The tool copies the selected APIs, API version sets, and revisions into the selected workspace. It appends a random hash to the API name and API suffix to avoid conflicts. 
+
+    For each detected API dependency:
```
Comment (mikebudzynski) [inline README.md:83]: We have to mention that we are changing the references to the migrated dependencies. For example, if a product is migrated, we will change the product assignment in a migrated API to point to the migrated product instead.

File: README.md [hunk 9]
```diff
@@ -1,24 +1,140 @@
...
+- [Azure CLI](https://learn.microsoft.com/cli/azure/install-azure-cli) for authentication 
+
+- An API Management instance with an existing [workspace](https://learn.microsoft.com/en-us/azure/api-management/how-to-create-workspace) 
+
+- Permissions
+    - Read access for API Management service-level APIs and [supported dependencies](#supported-dependencies) that can be migrated 
+    - Write access for APIs and all [supported dependencies](#supported-resource-types-and-dependencies) in the workspace
+
+## Run the tool
+
```
Comment (mikebudzynski) [inline README.md:42]: 1. Download / check out the tool from the releases section. The `main` branch may contain unstable or untested code.

**Review by mikebudzynski — CHANGES_REQUESTED** [review]

File: README.md [hunk 10]
```diff
@@ -1,24 +1,140 @@
...
+* Delete the original APIs and the dependencies that were marked for deletion. 
+
+* Remove the suffix in the API URL path and from the name of resources whose original versions were deleted from the service level. 
+
+## Log
+
+The log of all operations is recorded in file _________.txt. The log file includes all the requests made by the tool, including URL, headers, payload, and response status. 
 
 ## Telemetry
```
Comment (tomkerkhove) [inline README.md]: Let's push this right before license maybe? Majority of folks does not care about it

File: README.md (hunk 1 above)
Comment (dlepow) [reply README.md]: Updated as suggested

File: README.md (hunk 2 above)
Comment (dlepow) [reply README.md]: Updated as suggested.

File: README.md (hunk 3 above)
Comment (dlepow) [reply README.md]: Added dotnet build/run commands

File: README.md (hunk 4 above)
Comment (dlepow) [reply README.md]: Currently the name is just ".NET". AFAIK Windows is not required.

File: README.md (hunk 5 above)
Comment (dlepow) [reply README.md]: Changed as suggested

File: README.md (hunk 6 above)
Comment (dlepow) [reply README.md]: Changed as suggested

File: README.md (hunk 7 above)
Comment (dlepow) [reply README.md]: Updated as suggested.

File: README.md (hunk 9 above)
Comment (dlepow) [reply README.md:42]: I didn't mention a releases section - don't see it used currently. I provided `git clone` steps for now (from main).

File: README.md (hunk 10 above)
Comment (dlepow) [reply README.md]: Moved as suggested - thanks.

File: README.md (hunk 8 above)
Comment (dlepow) [reply README.md:83]: Added Note: 
"Note: Migrated APIs in a workspace are updated to reference any dependency that's migrated to the workspace. For example, if a dependent product is migrated, the product assignment in a migrated API is updated to point to the migrated product instead."

**Review by mikebudzynski — APPROVED** [review]

**Pull request merged**